src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java

```diff
@@ -94,7 +95,7 @@ private void handleEnvelope(
 
     private byte[] download(BlobClient blobClient) throws IOException {
         try (var outputStream = new ByteArrayOutputStream()) {
-            blobClient.download(outputStream);
+            blobClient.getBlockBlobClient().download(outputStream);
```

Comment:
I still think we should refactor this approach into `copy` functionality straight from one container to another

Reply:
definitely this can cause out of memory

Reply:
should be considered as a separate PR ;). let's fix this flow